prog/collide.go

```diff
@@ -97,3 +97,40 @@ func DoubleExecCollide(origProg *Prog, rand *rand.Rand) (*Prog, error) {
 	prog.Calls = append(prog.Calls, dupCalls...)
 	return prog, nil
 }
+
+// Duplicate every call in the program and execute the duplicated copies in parallel.
+// To ensure that async call execute in order, set async to every other call.
+// Example input:
+// r1 = test1()
+// test2(r1)
+// test3(r1)
+// Example output:
+// r1 = test1() (async)
+// r2 = test1()
+// test2(r1) # Note that we don't assign async here.
+// test2(r2)
+// test3(r1) (async)
+// test3(r1).
+func ParallelExecCollide(origProg *Prog, rand *rand.Rand) (*Prog, error) {
+	if len(origProg.Calls)*2 > MaxCalls {
```

Comment:
I would not return in this case, but instead add syscalls until we reach MaxCalls.
Say, if MaxCalls=20 and we have 11 syscalls, we still can get lots of collisions on this program.
However, if we have 19 syscalls, then it probably does not make sense to start duplicating them.
Maybe check we below, say, 75% of max size?

Reply:
I think it makes sense. Updated the code.